src/components/data-hooks.js

```diff
@@ -630,6 +630,36 @@ export function useRasterData(
       loaders[dataset].loaders.raster.load().catch(e => warn(e, setWarning)).then((payload) => {
         if (!payload) return;
         const { data, url: urls, coordinationValues } = payload;
+
+        /*
+        We need to merge the spatialRasterLayers properties because there are multiple
+        sub-properties that can be incomplete (for example colors without silders,
+        or just special selection choices).  Thus we merge the initial values into the (auto) values
+        from the `payload` and then null out the initial values so that `initCoordinationSpace`
+        sets the coordiantion value for raster layers.
+        */
+        // eslint-disable-next-line no-unused-expressions
+        if (initialCoordinationValues?.spatialRasterLayers) {
+          coordinationValues.spatialRasterLayers.forEach((layer, layerIndex) => {
+            const initialLayer = initialCoordinationValues.spatialRasterLayers[layerIndex];
+            const newChannels = layer.channels.map((channel, channelIndex) => ({
+              ...channel,
+              ...initialLayer.channels[channelIndex],
+            }));
+            // eslint-disable-next-line no-param-reassign
+            coordinationValues.spatialRasterLayers[layerIndex] = {
+              ...layer,
+              ...initialLayer,
+              channels: newChannels,
+            };
+          });
+        }
+        const newInitialCoordinationValues = {
+          ...(initialCoordinationValues || {}),
+          spatialRasterLayers: null,
+        };
```

Comment:
```suggestion
        if (initialCoordinationValues?.spatialRasterLayers) {
          initialCoordinationValues.spatialRasterLayers
            .forEach((initialLayer, initialLayerIndex) => {
              const layerIndex = initialLayer.index;
              const layer = coordinationValues.spatialRasterLayers
                .find(l => l.index === layerIndex);
              if (initialLayer && layer) {
                const newChannels = initialLayer.channels.map((initialChannel) => {
                  const channel = layer.channels
                    .find(c => equal(c.selection, initialChannel.selection));
                  return {
                    ...channel,
                    ...initialChannel,
                  };
                });
                // eslint-disable-next-line no-param-reassign
                initialCoordinationValues.spatialRasterLayers[initialLayerIndex] = {
                  ...layer,
                  ...initialLayer,
                  channels: newChannels,
                };
              }
            });
        }
        const newInitialCoordinationValues = initialCoordinationValues || coordinationValues;
```

This doesn't work if the layers or channels in the initial array are not in the default ordering, I think it needs  `.find()` to match on `layer.index` and `channel.selection`

Reply:
For example if I try this initial value for `spatialRasterLayers` with the Spraggins dataset:

```
"spatialRasterLayers": {
            "A": [{
                "type": "raster",
                "index": 1,
                "visible": true,
                "colormap": null,
                "opacity": 1,
                "domainType": "Min/Max",
                "transparentColor": null,
                "renderingMode": "Additive",
                "use3d": false,
                "channels": [{
                    "selection": {
                        "channel": 0
                    },
                    "color": [255, 0, 0]
                }]
            }]
        },
```

Reply:
Good catch.  I still needed to work out initialization of the channel statistics as well because they get initialized for the first four channels regardless  which is problematic.